### ISSUE TITLE
Infix operator error in jsx ternary

### ISSUE THREAD

**Issue by aabenoja** [eslint/eslint#3016]
I'm getting an `Infix operators must be spaced` error on the following ternary:

```jsx
let optionalComponent = this.props.someOptionalVal ? (
  <span>
    {this.props.someOptionalVal}
  </span>
) : null;
```

We have a couple of places where we use ternaries like this to choose whether to generate an optional item or assign a `null` in its place. There error is being thrown on the colon before `null`.

**Comment by michaelficarra** [comment]
Did you mean to file this against espree instead?

**Comment by Jackyjjc** [comment]
same here with `something.prop ? (d.something || 'blah') : d.something`

giving me Infix operators must be spaced. This is in node.js with the latest version of eslint installed from npm

**Comment by nzakas** [comment]
Thanks for the report. Can you please provide:

1. The exact version of ESLint you are using
2. The config for the rule
3. The complete ESLint output

**Comment by Jackyjjc** [comment]
The version i fall back to but it is still failing is eslint@0.22.1

```
parser: babel-eslint
environment:
  node: true
  mocha: true
ecmaFeatures:
  generators: true
  modules: true
  blockBindings: true
rules:
  semi: 2
  quotes: [2, "single"]
  max-len: [2, 120]
  no-undef: 0
  strict: 0
  no-extra-strict: 0
  no-use-before-define: 0
  space-unary-ops: 0
  curly: 0
  no-underscore-dangle: 0
  new-cap: [2, {"capIsNew": false}]
  new-parens: 0
  no-extend-native: 0
```

```
> eslint .

xxx/xxx/xxx.js
  78:76  error  Infix operators must be spaced  space-infix-ops

✖ 1 problem (1 error, 0 warnings)


npm ERR! Darwin 14.3.0
npm ERR! argv "node" "/Users/jchen/xxx/xxx/node_modules/.bin/npm" "run" "lint"
npm ERR! node v0.12.2
npm ERR! npm  v2.13.0
npm ERR! code ELIFECYCLE
npm ERR! lint: `eslint .`
npm ERR! Exit status 1
npm ERR!
npm ERR! Failed at the  lint script 'eslint .'.
npm ERR! This is most likely a problem with the xxx package,
npm ERR! not with npm itself.
npm ERR! Tell the author that this fails on your system:
npm ERR!    eslint .
npm ERR! You can get their info via:
npm ERR!     npm owner ls xxx
npm ERR! There is likely additional logging output above.
```

**Comment by aabenoja** [comment]
eslint v0.24.1

```
  "env":
    "browser": true,
    "node": true
  "ecmaFeatures":
    "jsx": true
  "parser": "babel-eslint",
  "plugins": ["react", "babel"],
  "rules":
    "constructor-super": 2,
    "comma-spacing": 2,
    "comma-style": [2, "last"],
    "one-var": [2, { "initialized": "never" }],
    "key-spacing": 0,
    "no-this-before-super": 2,
    "no-underscore-dangle": 0,
    "no-unused-vars": [2, { "vars": "all", "args": "none" }],
    "no-var": 2,
    "babel/object-shorthand": 2,
    "quotes": [2, "single", "avoid-escape"],
    "react/display-name": 0,
    "react/jsx-no-undef": 2,
    "react/jsx-uses-react": 2,
    "react/no-did-mount-set-state": 2,
    "react/no-did-update-set-state": 2,
    "react/no-multi-comp": 2,
    "react/prop-types": [2, { "ignore": [ "children", "className" ] }],
    "react/react-in-jsx-scope": 2,
    "react/self-closing-comp": 2,
    "react/wrap-multilines": 2,
    "react/jsx-uses-vars": 2,
    "strict": 0
```

```
> npm run lint && npm run build && npm run tests-set
> eslint ./

xxx/xxx.js
   56:6  error  Infix operators must be spaced  space-infix-ops
   62:6  error  Infix operators must be spaced  space-infix-ops
   68:6  error  Infix operators must be spaced  space-infix-ops
   74:6  error  Infix operators must be spaced  space-infix-ops
   90:6  error  Infix operators must be spaced  space-infix-ops
  104:6  error  Infix operators must be spaced  space-infix-ops
  112:6  error  Infix operators must be spaced  space-infix-ops
  133:6  error  Infix operators must be spaced  space-infix-ops
  147:6  error  Infix operators must be spaced  space-infix-ops

xxx/xxx.js
  31:6  error  Infix operators must be spaced  space-infix-ops

xxx/xxx/xxx.js
  179:10  error  Infix operators must be spaced  space-infix-ops

✖ 11 problems (11 errors, 0 warnings)

npm ERR! Linux 3.13.0-40-generic
npm ERR! argv "/home/travis/.nvm/versions/io.js/v2.3.4/bin/iojs" "/home/travis/.nvm/versions/io.js/v2.3.4/bin/npm" "run" "lint"
npm ERR! node v2.3.4
npm ERR! npm  v2.12.1
npm ERR! code ELIFECYCLE
npm ERR! react-bootstrap@0.23.7-docs.1 lint: `eslint ./`
npm ERR! Exit status 1
```

@michaelficarra Is most likely spot on. Comparing my build logs between this occurrence and our last successful build it's not just eslint being updated to v0.24.1, but espree getting updated to v2.2.0

**Comment by AlexKVal** [comment]
The same issue https://github.com/babel/babel-eslint/issues/149
I hope it'll add more info :cherries:

**Comment by hzoo** [comment]
I think you can close this since it's using `"parser": "babel-eslint",` -  it's an issue after updating babel to latest - 5.7.2. Will be fixed there.

**Comment by ilyavolodin** [comment]
Closing per @hzoo
